### ISSUE TITLE
pythran with gcc 8.0.1

### ISSUE THREAD

**Issue by nbecker** [serge-sans-paille/pythran#886]
Looks like an issue after upgrading to fedora 28, which includes gcc-8.0.1 (this code was working on fedora 27 with gcc-7.3.1)

~/.local/bin/pythran G.py
WARNING: Disabling color, you really want to install colorlog.
In file included from /usr/include/c++/8/unordered_map:46,
                 from /home/nbecker/.local/lib/python3.6/site-packages/pythran-0.8.5-py3.6.egg/pythran/pythonic/include/utils/shared_ref.hpp:6,
                 from /home/nbecker/.local/lib/python3.6/site-packages/pythran-0.8.5-py3.6.egg/pythran/pythonic/include/types/ndarray.hpp:9,
                 from /tmp/tmpbvoipygr.cpp:9:
/usr/include/c++/8/bits/hashtable.h: In instantiation of 'class std::_Hashtable<{anonymous}::pythonic::types::array<long int, 2>, std::pair<const {anonymous}::pythonic::types::array<long int, 2>, long int>, std::allocator<std::pair<const {anonymous}::pythonic::types::array<long int, 2>, long int> >, std::__detail::_Select1st, std::equal_to<{anonymous}::pythonic::types::array<long int, 2> >, std::hash<{anonymous}::pythonic::types::array<long int, 2> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<true, false, true> >':
/usr/include/c++/8/bits/unordered_map.h:105:18:   required from 'class std::unordered_map<{anonymous}::pythonic::types::array<long int, 2>, long int, std::hash<{anonymous}::pythonic::types::array<long int, 2> >, std::equal_to<{anonymous}::pythonic::types::array<long int, 2> >, std::allocator<std::pair<const 
{anonymous}::pythonic::types::array<long int, 2>, long int> > >'
/home/nbecker/.local/lib/python3.6/site-packages/pythran-0.8.5-py3.6.egg/pythran/pythonic/include/types/dict.hpp:105:57:   required from 'class {anonymous}::pythonic::types::dict<{anonymous}::pythonic::types::array<long int, 2>, long int>'
/usr/include/c++/8/bits/move.h:75:14:   required from 'constexpr _Tp&& std::forward(typename std::remove_reference<_Tp>::type&) [with _Tp = {anonymous}::pythonic::types::dict<{anonymous}::pythonic::types::array<long int, 2>, long int>; typename std::remove_reference<_Tp>::type = {anonymous}::pythonic::types::dict<{anonymous}::pythonic::types::array<long int, 2>, long int>]'
/home/nbecker/.local/lib/python3.6/site-packages/pythran-0.8.5-py3.6.egg/pythran/pythonic/functools/partial.hpp:62:3:   required by substitution of 'template<class ... Types> decltype ({anonymous}::pythonic::functools::partial((forward<Types>)(<unnamed>::pythonic::functools::functor::partial::operator()::types)...)) {anonymous}::pythonic::functools::functor::partial::operator()(Types&& ...) const [with Types = {__pythran_G::Decoding_logBP_lambda2, {anonymous}::pythonic::types::ndarray<double, 2>, {anonymous}::pythonic::types::dict<{anonymous}::pythonic::types::array<long int, 2>, long int>, long int}]'
/tmp/tmpbvoipygr.cpp:314:48:   required from 'struct __pythran_G::Decoding_logBP::type<{anonymous}::pythonic::types::ndarray<long int, 2>, {anonymous}::pythonic::types::list<{anonymous}::pythonic::types::list<long int> >, {anonymous}::pythonic::types::list<{anonymous}::pythonic::types::list<long int> >, {anonymous}::pythonic::types::ndarray<double, 2>, {anonymous}::pythonic::types::ndarray<double, 1>, long int>'
/tmp/tmpbvoipygr.cpp:697:259:   required from here
/usr/include/c++/8/bits/hashtable.h:189:21: error: static assertion failed: unordered container must have a non-const, non-volatile value_type

**Comment by nbecker** [comment]
clang++ gives the same error (uses the same header)

**Comment by serge-sans-paille** [comment]
Ma doué... what's the content of ``G.py`` ?

**Comment by nbecker** [comment]
You'll probably find this code is familiar:
https://gist.github.com/nbecker/290beaa74491b3c105a3ac9683203b36

**Comment by serge-sans-paille** [comment]
Yeah, I remember that one. It compiles correctly with
```
% g++-8 --version
g++-8 (Debian 8-20180414-1) 8.0.1 20180414 (experimental) [trunk revision 259383]
```
I'll try to reproduce

**Comment by serge-sans-paille** [comment]
Issue with gcc *8* should be fixed by #888. It's all about the 8 digit :-)

**Comment by nbecker** [comment]
Yes, that seems to fix it for me, thanks!

**Comment by serge-sans-paille** [comment]
Fixed in #888
